### ISSUE TITLE
Adding an extra node pool resets custom NSG rules

### ISSUE THREAD

**Issue by netiperher** [Azure/AKS#976]
**What happened**:
When running a AKS cluster with multiple node pools preview enabled and when adding an extra node pool all rules for the cluster NSG are reset. This means that any custom rules and rules created for load balancers are removed. The load balancer rules are eventually added back automatically.

**What you expected to happen**:
Custom NSG rules to be preserved when adding an extra node pool.

**How to reproduce it (as minimally and precisely as possible)**:
1. Create a AKS cluster with VMSS and advanced networking (Azure network plugin)
2. Add a second node pool to the cluster

```
az aks create --enable-vmss --kubernetes-version 1.12.7 --node-vm-size Standard_D8s_v3 --node-count 1  --network-plugin azure --vnet-subnet-id ...
az aks nodepool add --node-vm-size Standard_F8s --node-count 1 --vnet-subnet-id ...
```

**Environment**:
- Kubernetes version (use `kubectl version`): v1.12.7

**Comment by coreywagehoft** [comment]
I saw this same situation today on a cluster.  I noticed we did an update on a nodepool that had our externally accessible pods.
```
az aks nodepool update --update-cluster-autoscaler --min-count 1 --max-count 10 -g <resourceGroup -n pool1 --cluster-name <clusterName>
```
We deleted the pods that are responsible for ingress and the NSGs were not re-created.

AKS Details:
v0.40.0-aks
Kubernetes: 1.14.5

**Comment by celamb4** [comment]
+1 running into this issue.

**Comment by xizhamsft** [comment]
@coreywagehoft , @celamb4 , could you clarify the issue? We deployed a fix recently that will make sure NSG/RouteTable are not changed during agent pool create. But seems like you are saying that agent pool update will overwrite your NSG too? If so, could you provider detailed repro steps?

**Comment by coreywagehoft** [comment]
@xizhamsft when I performed an update to the node pool using the command above, it would complete the nodepool update.

I noticed that NSG rules that were created by the cloud-controller-manager for our external load balancer were removed.  We were using a basic load balancer at the time assigned to that specific nodepool.  We are currently moving to standard load balancers.

The only way I could get the NSGs to re-generate was to delete the Kubernetes Service that had the type LoadBalancer and recreate it.  Once I did that the NSGs were added back correctly.

To re-create this.  Have a nodepool with a Kubernetes Service with the type LoadBalancer assigned to the nodepool.  Attach the public basic load balancer to that scale set.  Then run the command I listed above on that node pool.  That will recreate the setup we had.
